sklearn/exceptions.py

```diff
@@ -135,7 +135,12 @@ class FitFailedWarning(RuntimeWarning):
     ...         gs.fit(X, y)  # This will raise a ValueError since C is < 0
     ...     except ValueError:
     ...         pass
-    ...     print(repr(w[-1].message))
+    ...     msg = repr(w[-1].message)
+    ...     # if msg.find('Traceback (most recent call last):\\n') > -1:
+    ...     if msg.count('\\n') > 1:  # has stacktrace in middle of string
```

Comment:
Having to include this here seems strange...

Also not sure why we want escaped newlines in the message rather than actually line breaks

Reply:
Totally agree format_exc is sufficient, did not realize about it and I thought you were just short abbreviating for format_exception, my mistake.  Take things literally :)

I agree it is strange.  As for the new lines, I kept getting complaints in ALL of the builds with `'\n'` so after checking the docs, it has to be escaped even in code blocks apparently.  But the errors must be some system specific stuff I had guessed `\r\n` or something like that but so far I have not figured out why 4 systems consistently error.

Reply:
But since we cannot put the stack trace into the expected output given that its easily change breakable from future refactorings or code which shifts line numbers, its best to filter the first and last line and compare those.  

Perhaps we need to use an even uglier `chr(13)` instead of `\\n` to avoid the errors and make the change all environment friendly.  I will give it a try if it continues to fail.

Reply:
It could also be a Unicode issue too, unfortunately this is a rather petty annoying detail but one that should be resolved.